src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -33,56 +33,113 @@ public PropertySubpatternCompletionProvider()
         {
         }
 
+        // Examples:
+        // is { $$
+        // is { Property.$$
+        // is { Property.Property2.$$
         public override async Task ProvideCompletionsAsync(CompletionContext context)
         {
             var document = context.Document;
             var position = context.Position;
             var cancellationToken = context.CancellationToken;
             var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
 
-            var token = TryGetOpenBraceOrCommaInPropertyPatternClause(tree, position, cancellationToken);
-            if (token == default || !(token.Parent.Parent is PatternSyntax))
+            var (propertyPatternClause, memberNameAccess) = TryGetPropertyPatternClause(tree, position, cancellationToken);
+            if (propertyPatternClause is null)
             {
                 return;
             }
 
             var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
-            var pattern = (PatternSyntax)token.Parent.Parent;
+            var pattern = (PatternSyntax)propertyPatternClause.Parent;
             var type = semanticModel.GetTypeInfo(pattern, cancellationToken).ConvertedType;
-            if (type == null)
+
+            if (memberNameAccess is not null)
+            {
+                // We have to figure out the type of the extended property ourselves, because
+                // the semantic model could not provide the answer we want in incomplete syntax:
+                // `c is { X. }`
+
+                type = GetMemberAccessType(type, memberNameAccess.Expression, document, semanticModel, position);
+            }
+
+            if (type is null)
             {
                 return;
             }
 
             // Find the members that can be tested.
-            IEnumerable<ISymbol> members = semanticModel.LookupSymbols(position, type);
-            members = members.Where(m => m.CanBeReferencedByName &&
-                IsFieldOrReadableProperty(m) &&
-                !m.IsImplicitlyDeclared &&
-                !m.IsStatic);
-
-            // Filter out those members that have already been typed
-            var propertyPatternClause = (PropertyPatternClauseSyntax)token.Parent;
+            var members = GetCandidatePropertiesAndFields(document, semanticModel, position, type);
 
-            // List the members that are already tested in this property sub-pattern
-            var alreadyTestedMembers = new HashSet<string>(propertyPatternClause.Subpatterns.Select(
-                // PROTOTYPE(extended-property-patterns) ExpressionColon
-                p => p.NameColon?.Name.Identifier.ValueText).Where(s => !string.IsNullOrEmpty(s)));
+            if (propertyPatternClause is not null)
```

Comment:
this test seems unnecessary (you checked it for null above).

Reply:
Thanks. It was the wrong test